crates/db_views/src/site_view.rs

```diff
@@ -2,19 +2,27 @@ use crate::structs::SiteView;
 use diesel::{result::Error, *};
 use lemmy_db_schema::{
   aggregates::structs::SiteAggregates,
-  schema::{site, site_aggregates},
+  schema::{site, site_aggregates, language, site_language},
   source::site::Site,
-};
+};use lemmy_db_schema::source::language::Language;
 
 impl SiteView {
   pub fn read_local(conn: &PgConnection) -> Result<Self, Error> {
-    let (mut site, counts) = site::table
-      .inner_join(site_aggregates::table)
-      .select((site::all_columns, site_aggregates::all_columns))
-      .order_by(site::id)
-      .first::<(Site, SiteAggregates)>(conn)?;
+    conn.build_transaction().read_write().run(|| {
+      let (mut site, counts) = site::table
+          .inner_join(site_aggregates::table)
+          .select((site::all_columns, site_aggregates::all_columns))
+          .order_by(site::id)
+          .first::<(Site, SiteAggregates)>(conn)?;
 
-    site.private_key = None;
-    Ok(SiteView { site, counts })
+      let languages = site_language::table
```

Comment:
Views should never have multiple queries, they should always be a single select.

The languages should come back from `GetSiteResponse`, probably next to `all_languages`, maybe `site_languages`.

Reply:
Done, named it `discussion_languages`.